src/actions/sources/blackbox.js

```diff
@@ -17,16 +17,45 @@ import type { ThunkArgs } from "../types";
 
 export function toggleBlackBox(source: Source) {
   return async ({ dispatch, getState, client, sourceMaps }: ThunkArgs) => {
-    const { isBlackBoxed, id } = source;
+    const { isBlackBoxed } = source;
 
     if (!isBlackBoxed) {
       recordEvent("blackbox");
     }
 
+    const startLocation = await sourceMaps.getGeneratedLocation(
+      { sourceId: source.id, line: 1, column: 0 },
+      source
+    );
+
+    const lines = (await sourceMaps.getOriginalSourceText(source)).text.split(
+      "\n"
+    );
+
+    let endLocation = { line: null };
+    let attempt = 0;
+    while (endLocation.line === null) {
```

Comment:
this is the hacky version of finding the end, let me know if anyone knows better way to do it.

Reply:
If this is moved to be part of the worker, we can use the SourceMapConsumer's `generatedPositionFor` with `bias: GREATEST_LOWER_BOUND` to do this automatic backward slide.

Reply:
awesome, will try that